client.go

```diff
@@ -435,6 +441,7 @@ func NewClient() *Client {
 		RetryMax:     defaultRetryMax,
 		CheckRetry:   DefaultRetryPolicy,
 		Backoff:      DefaultBackoff,
+		PrepareRetry: DefaultPrepareRetry,
```

Comment:
Since this default function does nothing, can we leave the default as `nil` here?